autonomous/autonomous.py

```diff
@@ -67,3 +70,114 @@ def engage_shooter(self):
         self.autoShoot.engage()
         self.shooter.engage()
         self.next_state("engage_shooter")
+
+
+
+
+class AutonomousAutoStart(AutonomousStateMachine):
+    """Creates the autonomous code"""
+    time = 1.4
+    MODE_NAME = "Hallway Autonomous"
+    DEFAULT = False
+    driveTrain: DriveTrain
+    shooter: ShooterLogic
+    goToDist: GoToDist
+    turnToAngle: TurnToAngle
+    pneumatics: Pneumatics
+    autoAlign: AutoAlign
+    autoShoot: AutoShoot
+    shooter: ShooterLogic
+    drive_speed = tunable(.25)
+    StopRunningFirstCall = True
+    CheckAngleFirstCall = True
+    TurnsCompleted = 0
+
+    @state(first = True)
+    def drive_forwards(self):
+        """Drives the bot forwards for 5 feet"""
+        self.driveTrain.setup()
+        self.driveTrain.resetDistTraveled()
+        self.goToDist.engage()
+        self.goToDist.start(-60)
+        self.driveTrain.execute()
+        self.next_state("stoprunning")
+
+    @state
+    def stoprunning(self):
+        self.goToDist.engage()
+        self.driveTrain.execute()
+        if self.StopRunningFirstCall:
+            self.StopRunningFirstCall = not self.StopRunningFirstCall
+            self.next_state("stoprunning")
+        elif self.goToDist.running:
+            self.next_state("stoprunning")
+        else:
+            self.next_state("turn")
+
+
+    @state
+    def turn(self):
+        """One method that completes all turns"""
+        turn1 = -90
+        turn2 = 180
+        turn3 = -90
+        self.turnToAngle.engage()
+        self.CheckAngleFirstCall = True
+        if self.TurnsCompleted == 0:
```

Comment:
Consider breaking these into their own states next time. So Turn1 is a state...and you continue in Turn1 until the turn is complete...once completed, self.next_state("turn2"),etc